### DISCUSSION TITLE
Is it possible to disable `update_root` for symlinks?

### DISCUSSION THREAD

**Discussion by jmoudrik** [nvim-tree/nvim-tree.lua#2888] [Q&A]
Hi!

Is it possible to disable symlinks from `update_focused_file.update_root` ? What I want is to keep `update_root=True` (so that sessions change root, etc), but without it following symlinks - the issue is that opening a symlink causes nvimtree with `update_focused_file.update_root.enabled = True` to "lose" the original root; I would rather it kept the symlink nested under the old tree.

I am now working around by putting the filename into the `ignore_list` manually, but that sucks. 

I believe the logic is here:
https://github.com/nvim-tree/nvim-tree.lua/blob/d43ab67d0eb4317961c5e9d15fffe908519debe0/lua/nvim-tree.lua#L35-L39
but neither path, nor nvims filetype show info about symlinks.

I think adding a custom function-like option into `ignore_list` would be best, but before implementing that, I even wonder if this is this the proper way to do it? Am I the only one who has symlinks && likes `update_root` or am I doing something wrong?

Thx!

**Comment by alex-courtis** [comment]
All the functionality that uses `change_root` does so based on the path of the buffer that nvim reports.

When `:e link-to-foo` the buffer name is `link-to-foo`. When selecting it in the tree, the buffer name is `foo`. That would result in the change in root.

A solution might be to open/edit links themselves rather than their target. That could be considered a bugfix and it's acceptable to nvim.

[pull requests](https://github.com/nvim-tree/nvim-tree.lua/blob/master/CONTRIBUTING.md) are always welcome and I'd be happy to help you out on that one.
